woocommerce/woocommerce-admin
# Add installed themes and WooCommerce support to theme browser

Fixes #2494

Adds installed themes to the theme browser as well as WC support error icons if the theme doesn't support WC.

### Questions/discussion

I initially looked at using `get_theme_support()` but this only works for the active theme.  Trying to include all themes and artificially activate this seems like it might cause a lot of trouble.

The current solution uses a regex match.  It's not perfect, but I think might do a decent job of detecting support.  It also puts the installed themes in the transient to prevent potential performance issues.

### Screenshots
<img width="932" alt="Screen Shot 2019-06-25 at 6 11 38 PM" src="https://user-images.githubusercontent.com/10561050/60090333-ec6bd080-9774-11e9-96ea-77031e4023c1.png">

### Detailed test instructions:

1. Get a local woocommerce.com test site running and use [this branch](https://github.com/Automattic/woocommerce.com/pull/6035).
2. Install Storefront or other wc supported themes.
3. Visit `wp-admin/admin.php?page=wc-admin#/?step=theme` in your local WP install.
4.  Make sure already installed themes show up in the list and are styled correctly (no "Live Demo" option and price/status line should state `Installed` or `Currently active theme`).
5.  Make sure Storefront does not show a lack of support error icon.

### Changelog Note:

Enhancement: Add installed themes to theme browser during onboarding.
